src/SwitchImplIntf.ml

```diff
@@ -0,0 +1,59 @@
+open Import
+
+module Lsp : sig
+  val switch : LanguageClient.t -> string -> string option Promise.t
+end = struct
+  let switch client fileName =
+    let targetFileName : string option Promise.t =
+      LanguageClient.sendRequest client "ocaml/didSwitchImplIntf"
+        (fileName : string)
+        ()
```

Comment:
If the language server will be able to support switching for file types other than the 6 listed in `Fallback`, we will need to figure out a way to send the language ID too.

I'm not completely familiar with the `vscode-languageclient` API, but I'm guessing that sending a value of type `Json.t` will work.

Reply:
> If the language server will be able to support switching for file types other than the 6 listed in Fallback, we will need to figure out a way to send the language ID too.

Could you elaborate on this? I'm not sure I'm getting the problem here.

> I'm not completely familiar with the vscode-languageclient API, but I'm guessing that sending a value of type Json.t will work.

That sounds right.

Reply:
For example, if a user requests to switch from `main.ml.in`, I imagine it would be helpful if the language server knew if it was set to the `ocaml` language ID by the user.

Passing the entire `document` is a good idea as it would give all the contextual information to the language server.

Reply:
Since `TextDocument.t`, which is now passed to ocaml-lsp in the request, is a type specific to vs code, wouldn't it make sense to pass just the filepath, a string or an alias to a string, to keep it as general as possible?

Reply:
Yes, I think you're right.

Reply:
Unless we establish a certain type that all clients of ocaml-lsp should follow; something like 

```ocaml
type params =
  { uri : string
  ; languageId : string
  }
```